### PR TITLE
Use std::mutex functions instead of the names in Threads::Mutex.

### PR DESCRIPTION
**Pull request by bangerth**
This then allows the deprecation of the Threads::Mutex member functions, as done in
the second commit.

In reference to #7228. See also #7232, #7235.

### CODE REVIEW

File: include/deal.II/base/thread_management.h [hunk 1]
```diff
@@ -309,7 +309,11 @@ namespace Threads
 
     /**
      * Acquire a mutex.
+     *
+     * This function is deprecated. Use the std::mutex::lock()
```
Comment (drwells) [inline include/deal.II/base/thread_management.h]: Could you add `@deprecated` here and below?

**Comment by drwells** [comment]
It looks like Travis found some trailing whitespace.

**Comment by drwells** [comment]
@bangerth I think that the timer class also uses POSIX functionality (for counting CPU time), so we are not completely rid of it if we just remove this class.

**Comment by bangerth** [comment]
Should all be fixed now!

**Pull request merged**